contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -236,90 +257,93 @@ contract MixinSignatureValidator is
         address walletAddress,
         bytes memory signature
     )
-        internal
+        private
         view
         returns (bool isValid)
     {
+        uint256 signatureLength = signature.length;
+        // Shave the signature type off the signature.
+        assembly {
+            mstore(signature, sub(signatureLength, 1))
+        }
+        // Encode the call data.
         bytes memory callData = abi.encodeWithSelector(
             IWallet(walletAddress).isValidSignature.selector,
             hash,
             signature
         );
+        // Restore the full signature.
         assembly {
-            let cdStart := add(callData, 32)
-            let success := staticcall(
-                gas,              // forward all gas
-                walletAddress,    // address of Wallet contract
-                cdStart,          // pointer to start of input
-                mload(callData),  // length of input
-                cdStart,          // write output over input
-                32                // output size is 32 bytes
-            )
-
-            switch success
-            case 0 {
-                // Revert with `Error("WALLET_ERROR")`
-                mstore(0, 0x08c379a000000000000000000000000000000000000000000000000000000000)
-                mstore(32, 0x0000002000000000000000000000000000000000000000000000000000000000)
-                mstore(64, 0x0000000c57414c4c45545f4552524f5200000000000000000000000000000000)
-                mstore(96, 0)
-                revert(0, 100)
-            }
-            case 1 {
-                // Signature is valid if call did not revert and returned true
-                isValid := mload(cdStart)
-            }
+            mstore(signature, signatureLength)
         }
-        return isValid;
+        // Static call the verification function.
+        (bool didSucceed, bytes memory returnData) = walletAddress.staticcall(callData);
```

Comment:
I believe that you can just do `IValidator.isValidSignature(...)` and it will compile with the `staticcall` opcode in Solidity ^0.5.0.

Reply:
Yeah, but don't we want to catch the revert?

Reply:
We can return the revert data, too.

Reply:
Implemented [here](../blob/dfb43ddc18ef10e9080a579e185e8ee7b8edfb33/contracts/exchange/contracts/src/MixinSignatureValidator.sol#L286) and [here](../blob/dfb43ddc18ef10e9080a579e185e8ee7b8edfb33/contracts/exchange/contracts/src/MixinSignatureValidator.sol#L344).